flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateOrAlterMaterializedTable.java

```diff
@@ -77,23 +82,25 @@ public SqlCreateMaterializedTable(
             SqlNodeList propertyList,
             SqlIntervalLiteral freshness,
             @Nullable SqlLiteral refreshMode,
-            SqlNode asQuery) {
-        super(OPERATOR, pos, false, false);
+            SqlNode asQuery,
+            boolean isOrAlter) {
+        super(isOrAlter ? CREATE_OR_ALTER_OPERATOR : CREATE_OPERATOR, pos, false, false);
         this.tableName = requireNonNull(tableName, "tableName should not be null");
-        this.tableConstraint = tableConstraint;
         this.comment = comment;
+        this.tableConstraint = tableConstraint;
         this.distribution = distribution;
         this.partitionKeyList =
                 requireNonNull(partitionKeyList, "partitionKeyList should not be null");
         this.propertyList = requireNonNull(propertyList, "propertyList should not be null");
         this.freshness = requireNonNull(freshness, "freshness should not be null");
         this.refreshMode = refreshMode;
         this.asQuery = requireNonNull(asQuery, "asQuery should not be null");
```

Comment:
If an `create or alter ` is an alter that just changes the refreshMode, or freshness, why does the asQuery need to be supplied? Could we delay validating this and only insist on asQuery being supplied if this really is a CREATE.